src/canvasObject.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var CanvasAnimationHelpers = require('./CanvasAnimationHelpers');
+
+var CanvasObject = function(config) {
+  this.needed = config.needed || false;
+  this.visible = config.visible || true;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.position = config.position || { x: 0, y: 0 };
+  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
+
+  this.id = config.canvas['@id'];
+  this.height = config.canvas.height;
+  this.width = config.canvas.width;
+  this.images = config.canvas.images;
+  this.label = config.canvas.label;
+  this.tileSourceUrl = config.canvas.images[0].resource.service['@id'] + '/info.json';
+};
+
+CanvasObject.prototype = {
+  openTileSource: function(x, y, width, viewer) {
```

Comment:
This should just be a swap, so it should already know the x, y, width, viewer, instead of having it passed in. 

Also, it should know whether the swap has already occurred (i.e. it's already a full tile source) and just return without doing an additional swap.